### PR TITLE
Date time format

### PR DESCRIPTION
**Pull request by kcarra**

### CODE REVIEW

**Comment by kcarra** [comment]
We're going to go back and refactor out the integration tests to dry up creating datasets with schemas.

File: apps/andi/assets/css/data_dictionary_form.scss [hunk 1]
```diff
@@ -15,21 +15,34 @@ $tick-height: 1rem + $field-padding - 0.44;
...
+      "selector . type-info"
+      "description description description"
       "pii masked demographic"
-      "biased rationale ."; 
- 
+      "biased rationale .";
+
       .select, .input {
-        width: 90%; 
+        margin: 10px 0 10px 0;
```
Comment (bbrewer-papa-x) [inline apps/andi/assets/css/data_dictionary_form.scss:27]: what's up with the pixels?

File: apps/andi/lib/andi/input_schemas/datasets/data_dictionary.ex [hunk 2]
```diff
@@ -102,4 +107,22 @@ defmodule Andi.InputSchemas.Datasets.DataDictionary do
   end
 
   defp validate_selector(changeset, _), do: changeset
+
+  defp validate_format(%{changes: %{type: type, format: format}} = changeset) when type in ["date", "timestamp"] do
+    case Formatter.validate(format) do
+      :ok ->
+        changeset
+
+      {:error, %RuntimeError{message: error_msg}} ->
```
Comment (bbrewer-papa-x) [inline apps/andi/lib/andi/input_schemas/datasets/data_dictionary.ex:116]: wait, that's what they return? ha, wtf

File: apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_field_editor.ex [hunk 3]
```diff
@@ -34,11 +35,29 @@ defmodule AndiWeb.EditLiveView.DataDictionaryFieldEditor do
...
+          <% field_type when field_type in ["date", "timestamp"] -> %>
+              <div class="format-label">
+                <%= label(@form, :format, "Format", class: "label label--required") %>
+                <a href="https://hexdocs.pm/timex/Timex.Format.DateTime.Formatters.Default.html" target="_blank">Help</a>
+              </div>
+              <%= text_input(@form, :format, id: id <> "_format", class: "data-dictionary-field-editor__format input") %>
+              <%= ErrorHelpers.error_tag(form_with_errors, :format) %>
+
+          <% _ -> %>
+             <%= "" %>
```
Comment (bbrewer-papa-x) [inline apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_field_editor.ex:56]: Wait, is this something we regularly do?

File: apps/andi/lib/andi/input_schemas/datasets/data_dictionary.ex (hunk 2 above)
Comment (cameronmarsh) [reply apps/andi/lib/andi/input_schemas/datasets/data_dictionary.ex:116]: Yeah it's weird

File: apps/andi/assets/css/data_dictionary_form.scss (hunk 1 above)
Comment (cameronmarsh) [reply apps/andi/assets/css/data_dictionary_form.scss:27]: Just matching what app.scss does for input fields

File: apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_field_editor.ex (hunk 3 above)
Comment (cameronmarsh) [reply apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_field_editor.ex:56]: No we should probably change this

Comment (bbrewer-papa-x) [reply apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_field_editor.ex:56]: Good

**Pull request merged**